infrastructure/tasks/15-functions_sync.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Run this task to sync Azure Functions staging slot
+ * to the source control repository code branch
+ * specified in the configuration file (`tfvars.json`).
+ * 
+ * This is equivalent to push the "Sync" button in the
+ * Azure portal -> Functions -> Deployments blade.
+ * 
+ * This task assumes that the following resources are already created:
+ *  - Resource group
+ *  - Functions (app service)
+ * 
+ */
+// tslint:disable:no-console
+// tslint:disable:no-any
+
+import { login } from "../../lib/login";
+
+import readConfig from "../../lib/config";
+const config = readConfig(__dirname + "/../tfvars.json");
+
+import webSiteManagementClient = require("azure-arm-website");
+
+export const run = async () => {
+  if (!config.functionapp_git_repo) {
+    return Promise.resolve();
+  }
+  const loginCreds = await login();
+
+  const webSiteClient = new webSiteManagementClient(
+    loginCreds.creds as any,
```

Comment:
please don't do that to me :)

Reply:
this comes from the Azure ARM APIs that contain "two types with the same name but unrelated", so there's no way to avoid this 'any' here